modules/checkm/qa/main.nf

```diff
@@ -0,0 +1,44 @@
+process CHECKM_QA {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::checkm-genome=1.2.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/checkm-genome:1.2.1--pyhdfd78af_0' :
+        'quay.io/biocontainers/checkm-genome:1.2.1--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(analysis_dir), path(marker_file), path(coverage_file)
+    path exclude_marker_file
+
+    output:
+    tuple val(meta), path("${prefix}.txt") , optional: true, emit: output
+    tuple val(meta), path("${prefix}.fasta"), optional: true, emit: fasta
+    path "versions.yml" , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args     = task.ext.args ?: ''
+    prefix       = task.ext.prefix ?: "${meta.id}"
```

Comment:
Def allows me to use `$prefix` outside of the `script` block - in this case in the output: block (otherwise it results in an error of 'referenced before assignment')